src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -4186,3 +4186,121 @@ TEST_CASE("BatchNormDeterministicTest", "[ANNLayerTest]")
   // The model should switch to training mode for predicting.
   REQUIRE(boost::get<BatchNorm<>*>(module.Model()[0])->Deterministic() == 0);
 }
+
+/**
+ * Linear module weight initialization test.
+ */
+TEST_CASE("LinearLayerWeightInitializationTest", "[ANNLayerTest]")
+{
+  size_t inSize = 10, outSize = 4;
+  Linear<>* linear = new Linear<>(inSize, outSize);
```

Comment:
Maybe I missed something but I think we can use `Linear<> linear(inSize, outSize)` as well, the idea is we don't have to manually deconstruct the object, which is problematic if one of the conditions fail.

Reply:
I do agree with Marcus. Pointers usually overcomplicate things. It's better to avoid them provided that they're not needed.